conftest.py

```diff
@@ -12,40 +12,18 @@
 """
 
 import os
-import os.path as osp
-import sys
-
-import pylsp_utils
 
 # ---- To activate/deactivate certain things for pytest's only
 # NOTE: Please leave this before any other import here!!
 os.environ['SPYDER_PYTEST'] = 'True'
 
-
-# ---- Detect if we're running in CI
-# Note: Don't import from spyder to keep this file free from local
-# imports.
-running_in_ci = bool(os.environ.get('CI'))
-
-
-# ---- Handle subrepos
-# NOTE: Please don't move this from here!
+# ---- Update PyLSP
 # When running in CI, subrepos are installed in the env.
-if not running_in_ci:
-    HERE = osp.dirname(osp.abspath(__file__))
-    DEPS_PATH = osp.join(HERE, 'external-deps')
-    i = 0
-    for path in os.listdir(DEPS_PATH):
-        external_dep_path = osp.join(DEPS_PATH, path)
-        sys.path.insert(i, external_dep_path)
-        i += 1
-
-    # Remove previous local PyLSP installation.
-    pylsp_utils.remove_installation()
-
-    # Install PyLS locally.
-    pylsp_utils.install()
-
+if not bool(os.environ.get('CI')):
```

Comment:
I'm pretty sure that os.environ['CI'] is a string "1" or "0". Should this not enforce bool?

Reply:
No, AFAIK; `bool("0")` is `True`, as `bool()` just uses the truthiness of the value rather than relying on implicit heuristics like that. You'd have to do `bool(int(...))` to get that behavior, but of course that doesn't work on non-ints.

AFAIK `CI` is only defined/non-empty on CI, so merely checking that it is defined and some non-empty string (IIRC, its`"True"`), as the suggestion does, is enough. The module-level constant is cast to bool in `conftest.py` merely for type correctness and for other consumers that may assume it has that literal value.